### PR TITLE
Started adding `Global` vs `Local` scope distinction

### PR DESCRIPTION
**Pull request by B-rando1**
### Main changes:
- I renamed the existing `ScopeSym` and all its derivatives to `VisibilitySym`, because it has to do with `public` vs `private` class members.  Hopefully this isn't too confusing...
- I created a new `ScopeSym` that deals with `global` vs `local` variables.
- I added `Scope` as a new parameter to `var`, `staticVar`, `extVar`, and `const`.

### Notes:
- Currently none of the renderers do anything with this, as they don't need to.  Julia will need it, though.
- To make our current code work with the changes, I had to give a scope to variables, constants, etc.  Because I wanted to get the code working again as soon as possible, I set  it to `local` in every instance.  I might regret this later, but I just haven't thought through what we need to do to give the proper value in each case.
- The order of the parameters for `var`, etc. could be up for debate.  Currently the `Scope` is the last parameter for these.  I think that makes sense, as it feels less important than the name or type of the variable.  As @JacquesCarette's [comment](https://github.com/JacquesCarette/Drasil/issues/3821#issuecomment-2212352023) shows, however, putting it nearer the start of the list of parameters would make certain internal functions look simpler.
    - I know there are several previous instances (e.g. [objMethodCall](https://github.com/JacquesCarette/Drasil/blob/a33094fd997e644590f9314c53c9986862880104/code/drasil-gool/lib/GOOL/Drasil/ClassInterface.hs#L667)) where we use a smart constructor to rearrange the order of parameters from a user-friendly ordering to an internal-friendly ordering.  Would this be a good place to do that?

Contributes towards #3821

### CODE REVIEW

**Review by JacquesCarette — CHANGES_REQUESTED** [review]
Just one thing to change, otherwise this looks great.

File: code/drasil-code/test/HelloWorld.hs [hunk 1]
```diff
@@ -27,77 +28,78 @@ description = "Tests various GOOL functions. It should run without errors."
...
+      (valueOf (var "b" int local) ?== litInt 5, helloIfBody)] helloElseBody, helloIfExists,
     helloSwitch, helloForLoop, helloWhileLoop, helloForEachLoop, helloTryCatch]])
 
 -- | Initialize variables used in the generated program.
 helloInitVariables :: (OOProg r) => MSBlock r
 helloInitVariables = block [comment "Initializing variables",
-  varDec $ var "a" int,
-  varDecDef (var "b" int) (litInt 5),
-  listDecDef (var "myOtherList" (listType double)) [litDouble 1.0,
+  varDec $ var "a" int local,
```
[the remark this answers is not in the document]
Comment (B-rando1) [reply code/drasil-code/test/HelloWorld.hs:38]: Good idea! I'll add that in.

While we're talking about smart constructors, do you think we should change the `var` in the typeclass to `var'`, and change the order of its parameters?  As you guessed, the fix for the type error in the C++ renderer (as well as a few places in `drasil-code`) would look nicer if we were to do that.  This is the current fix, for reference:
https://github.com/JacquesCarette/Drasil/blob/fbfc58e5df6d0fb2a6d7b7a19f5f19542fb4824d/code/drasil-gool/lib/GOOL/Drasil/LanguageRenderer/CppRenderer.hs#L275-L278
If we changed the operand order, we could get rid of the lambdas and leave the functions partially applied, as they were before.

[the remark this answers is not in the document]
Comment (JacquesCarette) [reply code/drasil-code/test/HelloWorld.hs:43]: Yes, changing the order of parameters to make things smooth is worthwhile.

**Comment by B-rando1** [comment]
I think we should find a better solution to my [comment](https://github.com/JacquesCarette/Drasil/issues/3821#issuecomment-2218049506) before we go further with this work.

**Review by JacquesCarette — APPROVED** [review]

**Pull request merged**